src/librustc_trans/intrinsic.rs

```diff
@@ -1193,11 +1193,16 @@ fn trans_gnu_try<'blk, 'tcx>(bcx: Block<'blk, 'tcx>,
         // managed by the standard library.
 
         attributes::emit_uwtable(bcx.fcx.llfn, true);
-        let catch_pers = match tcx.lang_items.eh_personality_catch() {
-            Some(did) => {
-                Callee::def(ccx, did, tcx.mk_substs(Substs::empty())).reify(ccx).val
+        let catch_pers = if cfg!(all(target_arch = "arm", not(target_os = "ios"))) {
```

Comment:
I think this needs to be looking at `sess` rather than `cfg!` (otherwise this'll always return false)